### DISCUSSION TITLE
searchBox widget default search text

### DISCUSSION THREAD

**Discussion by mikebhart** [algolia/instantsearch#4517] [General]
I'm using the searchBox widget. As you enter the page, is there a way I can automatically add default search text into the input field with the search results rendered.

**Comment by Haroenv** [comment]
You can use `initialUiState: { myIndex: { query: 'myQuery' } }` or enable it via routing:

- https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#widget-param-initialuistate
- https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/js/

**Comment by mikebhart** [comment]
Thankyou very much.
